From c90c891e89844efd0e0ecb8dce709d1c9c81d20e Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob2.chen@rock-chips.com>
Date: Mon, 18 Dec 2017 18:17:50 +0800
Subject: [PATCH 2/4] DEBIAN-ONLY: use mmap64

---
 osal/allocator/allocator_drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/osal/allocator/allocator_drm.c b/osal/allocator/allocator_drm.c
index 48735c9..1411941 100644
--- a/osal/allocator/allocator_drm.c
+++ b/osal/allocator/allocator_drm.c
@@ -68,7 +68,7 @@ static inline void *drm_mmap(void *addr, size_t length, int prot, int flags,
 
 /* assume large file support exists */
 #  define drm_mmap(addr, length, prot, flags, fd, offset) \
-              mmap(addr, length, prot, flags, fd, offset)
+              mmap64(addr, length, prot, flags, fd, offset)
 
 #  define drm_munmap(addr, length) \
               munmap(addr, length)
-- 
2.15.1

