From 61087490360e06f57ad9df52de324867b6e709cd Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob2.chen@rock-chips.com>
Date: Thu, 7 Dec 2017 13:07:25 +0800
Subject: [PATCH 5/6] TMP-FIX: mppvideodec: add hvc1, hev1 for h265

Signed-off-by: Jacob Chen <jacob2.chen@rock-chips.com>
---
 gst/rockchipmpp/gstmppvideodec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/rockchipmpp/gstmppvideodec.c b/gst/rockchipmpp/gstmppvideodec.c
index 315dddf..f1c39bd 100644
--- a/gst/rockchipmpp/gstmppvideodec.c
+++ b/gst/rockchipmpp/gstmppvideodec.c
@@ -48,7 +48,7 @@ static GstStaticPadTemplate gst_mpp_video_dec_sink_template =
         "parsed = (boolean) true"
         ";"
         "video/x-h265,"
-        "stream-format = (string) { byte-stream },"
+        "stream-format = (string) { hvc1, hev1, byte-stream },"
         "alignment = (string) { au },"
         "parsed = (boolean) true"
         ";"
-- 
2.15.1

