From ee24deb626383633f09f47409333832670c65c10 Mon Sep 17 00:00:00 2001
From: Jacob Chen <jacob2.chen@rock-chips.com>
Date: Thu, 7 Dec 2017 13:11:10 +0800
Subject: [PATCH 6/6] DEBIAN: update rules for release_20171218

Signed-off-by: Jacob Chen <jacob2.chen@rock-chips.com>
---
 debian/changelog | 33 +++++++++++++++++++++++++++++++++
 debian/rules     |  2 +-
 2 files changed, 34 insertions(+), 1 deletion(-)

diff --git a/debian/changelog b/debian/changelog
index 2017162..e9830a8 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -1,3 +1,36 @@
+gstreamer1.0-rockchip (20171218-1) UNRELEASED; urgency=medium
+
+  [ ayaka ]
+  * rockchipmpp: keep input metadata for decoder
+
+  [ Randy Li ]
+  * rockchipmpp: allocate the correct buffer size for NV16
+  * rockchipmpp: support more formats for encoder
+  * build: fix some error in build rules
+
+  [ ayaka ]
+  * rockchipmpp: keep input metadata for decoder
+  * tests: examples: add more features to ayamero
+
+  [ Randy Li ]
+  * tests: move the playing operation
+  * rockchipmpp: increase the default quality settings
+  * rockchipmpp: add support for MPEG-1/2/4
+  * rockchipmpp: place the caps of the src port in a safe way
+  * rockchipmpp: only mapping the buffer when it is needed
+
+  [ Jacob Chen ]
+  * DEBIAN-ONLY: rockchipmpp: adaptations to 1.10
+  * DEBIAN-ONLY: rockchipmpp: allow jpeg
+  * DEBIAN-ONLY: add timeout
+  * DEBIAN-ONLY: rockchipmpp: always use simple io mode
+  * TMP-FIX: enc: correct sps order
+  * TMP-FIX: enc: fix invaild data
+  * TMP-FIX: mppvideodec: add hvc1, hev1 for h265
+  * DEBIAN: update rules
+
+ -- Jacob Chen <jacob2.chen@rock-chips.com>  Mon, 18 Dec 2017 17:01:21 +0800
+
 gstreamer1.0-rockchip (1.10-5) testing; urgency=medium
 
   [ Jacob Chen ]
diff --git a/debian/rules b/debian/rules
index 974e090..af8f9ec 100755
--- a/debian/rules
+++ b/debian/rules
@@ -21,5 +21,5 @@
 # This is example for Cmake (See https://bugs.debian.org/641051 )
 override_dh_auto_configure:
 	NOCONFIGURE=true ./autogen.sh
-	dh_auto_configure -- #	-DCMAKE_LIBRARY_PATH=$(DEB_HOST_MULTIARCH)
+	dh_auto_configure -- --disable-rkximage #	-DCMAKE_LIBRARY_PATH=$(DEB_HOST_MULTIARCH)
 
-- 
2.15.1

